lib/subjects/postgresql_selection.rb

```diff
@@ -23,13 +23,50 @@ def selection_strategy
     def available
       query = Subjects::SetMemberSubjectSelector.new(workflow, user).set_member_subjects
       subject_set_ids = if workflow.grouped
-                            # respect the user if they want to select from a training set
-                            opts[:subject_set_id]
-                          else
-                            # default mode: do not select from training sets
-                            workflow.non_training_subject_sets.pluck(:id)
-                          end
-      query.where(subject_set_id: subject_set_ids)
+                          # respect the user if they want to select from a training set
+                          Array.wrap(opts[:subject_set_id])
+                        else
+                          # default mode: do not select from training sets
+                          workflow.non_training_subject_sets.pluck(:id)
+                        end
+      if Gem::Version.new(Rails.version) > Gem::Version.new('5.0')
```

Comment:
we could look at only running this code for rails 5.0 and allow 5.1 to run the old code?
```suggestion
      if Gem::Version.new(Rails.version) > Gem::Version.new('5.0') && Gem::Version.new(Rails.version) < Gem::Version.new('5.1')
```